sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -2837,4 +2837,13 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
         "parameter" -> toSQLId("unit"),
         "invalidValue" -> s"'$invalidValue'"))
   }
+
+  def invalidBooleanStatementError(stmt: String): Throwable = {
```

Comment:
let's use `SqlScriptingErrors` class for this exception. 
you can check this PR: https://github.com/apache/spark/pull/47553, where we introduced `SqlScriptingErrors` and `SqlScriptingException` classes for more cleaner messages (including the line info at the beginning etc).
these classes require `Origin` to be provided, but in this case `Origin` already exists within `SingleStatementExec` so it should be quite simple.

Reply:
Okay.